woocommerce/woocommerce
# Add coming soon image gallery pattern

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/53302.

This adds an image gallery pattern to the coming soon template, allowing users to showcase a preview of what they are selling with a variety of small images.

It is based off the T24 homepage pattern "[Portfolio home template with post featured images](https://github.com/WordPress/wordpress-develop/blob/b8ecfbf9542198b1d88230221faeca6b71fa0ac7/src/wp-content/themes/twentytwentyfour/patterns/template-home-portfolio.php#L4)", which displays blog posts featured images. I have replaced those with image blocks to allow the user to easily change the sample images with their own.

A follow-up PR will explore using existing images in the product catalogue to make it even simpler to use.

<img width="1792" alt="Screen Shot 2024-12-21 at 12 11 51 am" src="https://github.com/user-attachments/assets/7cd1696e-b94f-41e2-8ecf-ddc3ad2e283b" />



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site with  coming-soon-newsletter-template feature flag enabled.
1. Enable coming soon mode in site visibility settings.
1. Go to Appearance > Editor > Templates > Coming soon
1. Choose Coming Soon Image Gallery design on the right sidebar (Template > Design > Coming Soon Image Gallery)
1. Confirm the layout matches the design in Figma.
1. Confirm that you can customize the styles, replace images, change the content, etc.
1. Select Coming Soon Image Gallery  Image template again and save the template.
1. Go to the site front end incognito tab to make sure the template is loaded correctly. The page should be responsive.
1. Go to Appearance > Editor > Templates and confirm the coming soon template preview shows the selected template.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
